src/impl_/pyclass.rs

```diff
@@ -1284,19 +1398,36 @@ impl<ClassT: PyClass, FieldT: ToPyObject, Offset: OffsetCalculator<ClassT, Field
 pub trait PyO3GetField: IntoPy<Py<PyAny>> + Clone {}
 impl<T: IntoPy<Py<PyAny>> + Clone> PyO3GetField for T {}
 
-/// Base case attempts to use IntoPy + Clone, which was the only behaviour before PyO3 0.22.
+/// IntoPy + Clone fallback case, which was the only behaviour before PyO3 0.22.
+impl<ClassT, FieldT, Offset>
+    PyClassGetterGenerator<ClassT, FieldT, Offset, false, false, true, false, false>
+where
+    ClassT: PyClass,
+    Offset: OffsetCalculator<ClassT, FieldT>,
+    FieldT: PyO3GetField,
+{
+    pub const fn generate(&self, name: &'static CStr, doc: &'static CStr) -> PyMethodDefType {
+        PyMethodDefType::Getter(PyGetterDef {
+            name,
+            meth: pyo3_get_value::<ClassT, FieldT, Offset>,
+            doc,
+        })
+    }
+}
+
+/// Base case attempts to use IntoPyObject + Clone
 impl<ClassT: PyClass, FieldT, Offset: OffsetCalculator<ClassT, FieldT>>
-    PyClassGetterGenerator<ClassT, FieldT, Offset, false, false>
+    PyClassGetterGenerator<ClassT, FieldT, Offset, false, false, false, false, false>
```

Comment:
Hmm, I think this would only make it a bit better if the const generics would be fixed for every case. Since we would have to pass them through for some cases, the readability does not really get any better IMO.